src/server/pps/server/pipeline_controller.go

```diff
@@ -787,28 +788,28 @@ func (pc *pipelineController) deletePipelineKubeResources() (retErr error) {
 
 	// Delete any services associated with pc.pipeline
 	selector := fmt.Sprintf("%s=%s", pipelineNameLabel, pc.pipeline)
-	opts := &metav1.DeleteOptions{
+	opts := metav1.DeleteOptions{
 		OrphanDependents: &falseVal,
 	}
-	services, err := kubeClient.CoreV1().Services(namespace).List(metav1.ListOptions{LabelSelector: selector})
+	services, err := kubeClient.CoreV1().Services(namespace).List(pc.ctx, metav1.ListOptions{LabelSelector: selector})
```

Comment:
We probably don't want to just take the stored context here.  It would be nice to actually have control over how long `deletePipelineKubeResources` can take before it times out.  That might end up being the stored context, at a level up.

Reply:
yeah I agree with this. I'm actually making some changes to the pipeline controller separately and would like to handle this in a separate PR if that seems alright